sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java

```diff
@@ -275,9 +306,33 @@ abstract static class Builder<T> {
      * @param connectionFactory The JMS {@link ConnectionFactory}.
      * @return The corresponding {@link JmsIO.Read}.
      */
+    @Override
```

Comment:
shall we note in javadoc that "myConnectionFactory" has to be serializable? If not then need to use withConnectionFactoryProviderFn